superduperdb/ext/unstructured/encoder.py

```diff
@@ -0,0 +1,66 @@
+import dataclasses as dc
+import pickle
+import typing as t
+
+from unstructured.documents.elements import Element
+from unstructured.partition.auto import partition
+from unstructured.partition.html import partition_html
+
+from superduperdb.components.encoder import Encoder
+
+
+def link2element(link, unstructure_kwargs):
+    if link.startswith("file://"):
+        link = link[7:]
+
+    if link.startswith("http"):
+        # Special handling is required for http
+        elements = partition_html(url=link, **unstructure_kwargs)
+    else:
+        elements = partition(link, **unstructure_kwargs)
+    return elements
+
+
+def create_encoder(unstructure_kwargs):
+    def encoder(x: t.Union[str, t.List[Element]]):
+        if isinstance(x, str):
+            elements = link2element(x, unstructure_kwargs)
+        elif isinstance(x, list) and isinstance(x[0], Element):
+            elements = x
+        else:
+            raise TypeError(f"Cannot encode {type(x)}, must be str or list of Element")
+        return pickle.dumps(elements)
+
+    return encoder
+
+
+def create_decoder(unstructure_kwargs):
+    def decoder(b: bytes):
+        if isinstance(b, str):
```

Comment:
b: bytes? 
how can it be `str`

Reply:
If we use encoder(uri=file_path), the uri(a `str`) will be passed in here

What do you think if I ban the use of `uri=xxx` format?

Because our downloader processes uri, it is currently not very compatible with `unstructured`

Reply:
I see..

Reply:
Done